src/client/graphics/layers/RadialMenu.ts

```diff
@@ -267,7 +267,36 @@ export class RadialMenu implements Layer {
   }
 
   tick() {
-    // Update logic if needed
+    // Only update when menu is visible
+    if (this.isVisible && this.clickedCell) {
+      const myPlayer = this.g
+        .playerViews()
+        .find((p) => p.clientID() == this.clientID);
+      if (myPlayer && myPlayer.isAlive()) {
+        const tile = this.g.ref(this.clickedCell.x, this.clickedCell.y);
+        myPlayer.actions(tile).then((actions) => {
```

Comment:
If you make the function `async`, you can use `await` here, further flattening the function:
```suggestion
        const actions = await myPlayer.actions(tile);
```